File: app/src/main/java/com/spidroid/starry/ui/messages/MessageContextMenuListener.java
package com.spidroid.starry.ui.messages;

import com.spidroid.starry.models.ChatMessage; // تأكد من وجود هذا الاستيراد إذا لم يكن موجودًا

public interface MessageContextMenuListener {
  void onReplySelected();

  void onEditSelected();

  void onDeleteSelected();

  void onReactionSelected();

  void onReportSelected(ChatMessage message); // تم إضافة بارامتر ChatMessage هنا
}

File: app/src/main/java/com/spidroid/starry/ui/messages/MessageContextMenuListener.kt
package com.spidroid.starry.ui.messages

import com.spidroid.starry.models.ChatMessage

// تأكد من وجود هذا الاستيراد إذا لم يكن موجودًا
interface MessageContextMenuListener {
    fun onReplySelected()

    fun onEditSelected()

    fun onDeleteSelected()

    fun onReactionSelected()

    fun onReportSelected(message: ChatMessage?) // تم إضافة بارامتر ChatMessage هنا
}